research-software-directory/RSD-as-a-service
# 955 improve performance

# Improve RSD v2 performance

Closes #955

Changes proposed in this pull request:
* loading mentions on the software and project page api is changed to more performant api requests. (reduced from ~300ms to ~50ms)
* auth endpoint response is improved to cache the list of login providers on initial load instead of reading it on each request (reduced from ~500ms to ~10ms after initial request).
* fix cumulative layout shift on software overview page when highlights section is used
* bump the node version to v20 in the fe image. Node v20 is going to be LTS version from October 2023, so we should switch from v18 to v20 asap. Advice to FE developers, use [nvm](https://github.com/nvm-sh/nvm) to manage multiple node versions on your machine. 
* apply fixes advised by Lighthouse audit: increase image cache to 1yr and provide width and height for images

How to test:
* The improvement cannot be tested without comparing it to performance of the previous version.
* `make start` to rebuild app
* Navigate between home, software, organisation and project pages. The loading of the pages should be improved.
* After merging this PR I will set this version on `dev server` and perform load testing. Then we can compare the results with the load test results of the previous version. The expectation is that performance should be improved by these changes. I will post the results of load test in rsd-v2-release PR.

PR Checklist:
